source/common/http/utility.cc

```diff
@@ -185,9 +185,7 @@ bool Utility::isInternalRequest(const HeaderMap& headers) {
 
 bool Utility::isWebSocketUpgradeRequest(const HeaderMap& headers) {
   return (headers.Connection() && headers.Upgrade() &&
-          (0 == StringUtil::caseInsensitiveCompare(
-                    headers.Connection()->value().c_str(),
-                    Http::Headers::get().ConnectionValues.Upgrade.c_str())) &&
+          (NULL != strcasestr(headers.Connection()->value().c_str(), Http::Headers::get().ConnectionValues.Upgrade.c_str())) &&
```

Comment:
Don't we still want this to be case insensitive? Maybe we should also do a split operation on `,`, it would be more robust, at the expense of performance.

Reply:
I was thinking the same thing about splitting on comma.  I was looking for a helper/utility that would do this, and I don't see one (which surprised me).  

Perhaps we can add a method to HeaderString for bool entryContains(const LowerCaseString& str) const;

Reply:
@ggreenway we have https://github.com/envoyproxy/envoy/blob/master/source/common/common/utility.h#L131 and when we gain absl in #1974 we could also use https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_split.h

Reply:
In this case, do you think the perf cost will be too high to split it and return it?  It would be faster to search in place, but the interface would be less-nice and less-reusable.

Reply:
I agree that in general we shouldn't copy and split on the hot path when serving reqs. In this particular case, it might be OK to do this perhaps, since we already know we're most likely doing an upgrade due to the earlier test for the Upgrade: header, and if we're doing a Websockets upgrade, it's probably fine to pay a tiny bit extra for what is presumably a long lived session ¯\_(ツ)_/¯

Reply:
I'm fine with that.  If we profile and find we're spending too much time there, we can fix it then.

Reply:
yeah in this case because we are already guarded by the existence of the upgrade header I think it's fine to split after that fact.

Reply:
Strcasestr is case insensitive. The question here is whether to abstract this into stringutil or not. This particular case, given that the contributor is doing a one off contribution, I am reluctant to insist on him debugging more :). I can do the follow up and clean it up as it’s not bad as is. But I leave it to you guys to decide on this.